__tests__/users/App.test.js

```diff
@@ -89,4 +89,53 @@ describe('App Component', () => {
     const url = await page.url();
     expect(url).toContain('?tab=verified');
   });
+
+  it('should update the URL query string on search', async () => {
+    const initialUrl = await page.url();
+
+    await page.waitForSelector('.search_field');
+
+    await page.type('.search_field', 'John Doe');
+    await page.click('.search_button');
+
+    const updatedUrl = await page.url();
+
+    expect(updatedUrl).not.toEqual(initialUrl);
+  });
+
+  it('should display user details when a user card is clicked', async () => {
+    await page.waitForSelector('.active_tab');
+
+    await page.click('.active_tab');
+
+    await page.waitForSelector('.user_details_section');
+
+    const userDetailsDisplayed =
+      (await page.$('.user_details_section')) !== null;
+
+    expect(userDetailsDisplayed).toBeTruthy();
+  });
+
+  it('should display search results matching the search term', async () => {
+    await page.type('.search_field', 'amit');
+    await page.click('.search_button');
+
```

Comment:
Question: how is a search being done here in testing, 
when the name is sent, does it make an API call, yes then where?

Reply:
No, its not making an api call. But i changed the logic for test case now. After performing search it now checks if the page contains a result containing the search term.